scripts/deleteExpiredAccessTokens.js

```diff
@@ -0,0 +1,179 @@
+    options.accessTokens = [];
+    options.parsedCouchDbUrl = url.parse(options.couchDbUrl);
+    options.postOptions = {
+        hostname: options.parsedCouchDbUrl.hostname,
+        port: options.parsedCouchDbUrl.port,
+        path: "/gpii/_bulk_docs",
+        auth: options.parsedCouchDbUrl.auth,
+        method: "POST",
+        headers: {
+            "Accept": "application/json",
+            "Content-Length": 0,          // IMPORTANT: FILL IN PER REQUEST
+            "Content-Type": "application/json"
+        }
+    };
+    fluid.log("COUCHDB_URL: '" +
+        options.parsedCouchDbUrl.protocol + "//" +
+        options.parsedCouchDbUrl.hostname + ":" +
+        options.parsedCouchDbUrl.port +
+        options.parsedCouchDbUrl.pathname + "'"
+    );
+    return options;
+};
+
+/**
+ * Create the step that retrieves access tokens from the database:  either
+ * only the expired tokens, or all of them.
+ * @param {Object} options - Access tokens URL and whether to filter:
+ * @param {Array} options.accessTokensUrl - The url for retrieving all of the
+ *                                          access tokens in the database.
+ * @param {Boolean} options.deleteAll - Flag indicating whether to ignore the
+ *                                      expiration date of the access tokens.
+ * @return {Promise} - A promise that resolves retrieving the tokens.
+ */
+gpii.accessTokens.retrieveExpiredAccessTokens = function (options) {
+    var details = {
+        requestUrl: options.accessTokensUrl,
+        requestErrMsg: "Error retrieving access tokens from the database: ",
+        responseDataHandler: gpii.accessTokens.filterExpiredAccessTokens,
+        responseErrMsg: "Error retrieving access tokens from database: "
+    };
+    return gpii.dbRequest.configureStep(details, options);
+};
+
+/**
+ * Given all the access tokens from the database, filter out only the ones that
+ * have expired and store them in an array.  If the "options" argument's
+ * "deleteAll" flag is set, no filtering is done and all of the access tokens are
+ * returned.
+ * @param {String} responseString - the response from the request to get all the
+ *                                  acess tokens.
+ * @param {Object} options - Where to store the to-be-deleted access tokens and
+ *                           whether to delete regardless of time of expiration:
+ * @param {Array} options.accessTokens - The access tokens sought.
+ * @param {Boolean} options.deleteAll - Whether to ignore time of deletion.
+ * @return {Array} - The access tokens.
+ */
+gpii.accessTokens.filterExpiredAccessTokens = function (responseString, options) {
+    if (options.deleteAll) {
+        fluid.log("Deleting all access tokens...");
+    } else {
+        fluid.log("Filtering for expired access tokens...");
+    }
+    var tokens = JSON.parse(responseString);
+    var expiredTokens = [];
+    options.totalTokens = 0;
+    if (tokens.rows) {
+        fluid.each(tokens.rows, function (aRow) {
```

Comment:
It would save a round of iteration by incorporating the 2nd loop at line 151, which marks the "deletion" flag, into this loop since the 2nd one is dealing with filtered access tokens anyway.

Reply:
I've looked and it appears that `gpii.dbRequest.markDocsForDeletion` is used only by this loop in `deleteExpiredAccessTokens.js`.  What do you think about simply removing the `markDocsForDeletion()` function from the `gpii.dbRequest` namespace?

Reply:
I think it's a good idea.